### PR TITLE
Move to C++20 for non-MSVC compilers

### PR DESCRIPTION
**Pull request by Tilka**
Just checking the builders. Windows will probably fail?

### CODE REVIEW

**Comment by shuffle2** [comment]
well...can you think of a way to fix it such that msvc and gcc are happy?

**Comment by shuffle2** [comment]
seems to have been fixed upstream: https://github.com/fmtlib/fmt/commit/3e28dc021cf38b14ac0e2b1a046f783e094bfffc

**Comment by shuffle2** [comment]
might be interesting to see if these 2 deprecation suppressions can be removed if non-msvc builds get on c++20: https://github.com/dolphin-emu/dolphin/blob/master/Source/VSProps/Base.props#L74

File: Externals/fmt/include/fmt/format.h [hunk 1]
```diff
@@ -2198,7 +2198,7 @@ FMT_CONSTEXPR20 auto do_write_float(OutputIt out, const DecimalFP& fp,
     -> OutputIt {
   auto significand = fp.significand;
   int significand_size = get_significand_size(fp);
-  static constexpr Char zero = static_cast<Char>('0');
+  const Char zero = static_cast<Char>('0');
```
Comment (Zopolis4) [inline Externals/fmt/include/fmt/format.h:2201]: Shouldn't this be part of a separate PR updating the fmt external to at least `3e28dc021cf38b14ac0e2b1a046f783e094bfffc`?

Comment (Tilka) [reply Externals/fmt/include/fmt/format.h:2201]: We should be using a git submodule but I was lazy.

Comment (shuffle2) [reply Externals/fmt/include/fmt/format.h:2201]: We have patches on fmt that would need to be taken care of upstream

**Comment by shuffle2** [comment]
poke on this?

**Comment by lioncash** [comment]
Is this good to go? I'd love to slam the merge button on this

**Comment by Tilka** [comment]
Good from my side.

**Review by lioncash — APPROVED** [review]
alright then, allons y (finally)
